demo/frameworks/angular2/index.htm

```diff
@@ -0,0 +1,77 @@
+
+<head>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-material-design/4.0.2/bootstrap-material-design.css" />
+    <link rel="stylesheet" href="../../common/index.css" />
+    <script src="https://npmcdn.com/core-js@2.4.1/client/shim.min.js"></script>
+    <script src="https://npmcdn.com/zone.js@0.8.12?main=browser"></script>
+    <script src="https://npmcdn.com/reflect-metadata@0.1.10"></script>
+    <script src="https://npmcdn.com/rxjs@5.4.1/bundles/Rx.min.js"></script>
+    <script src="https://npmcdn.com/@angular/core@4.2.3/bundles/core.umd.js"></script>
+    <script src="https://npmcdn.com/@angular/common@4.2.3/bundles/common.umd.js"></script>
+    <script src="https://npmcdn.com/@angular/compiler@4.2.3/bundles/compiler.umd.js"></script>
+    <script src="https://npmcdn.com/@angular/platform-browser@4.2.3/bundles/platform-browser.umd.js"></script>
+    <script src="https://npmcdn.com/@angular/platform-browser-dynamic@4.2.3/bundles/platform-browser-dynamic.umd.js"></script>
+    <script src="../../../dist/xcomponent.js"></script>
+    <script src="./login.js"></script>
+</head>
+
+<body>
+    <my-app>Loading...</my-app>
+    <script>
+
+        // component
+        (function(app) {
+            app.AppComponent =
+                ng.core.Component({
+                selector: 'my-app',
+                template: `
+                        <div>
+                            <h3>Log in on xyz.com</h3>
+                            <div id="container">
+                                <my-login-component [props]="props"></my-login-component>
```

Comment:
I haven't played around with event-emitters yet (more abstractions over vanilla functions seems like an anti-pattern to me) but for sure, let's see how well this could fit in especially with loose props.